### PR TITLE
[extension/healthcheckv2] Add skeleton

### PR DESCRIPTION
**Pull request by mwear**
**Description:** 
This PR is an attempt to decompose https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30673 into more manageable pieces for review. It introduces the skeleton and config for the new healthcheck extension. The extension itself is implemented as multiple subcomponents. That is, the http and grpc services implement the component interface individually, as does the extension that ties them together. I mention this because this PR introduces the config structs and this is relevant for understanding why it is organized as it is. Additionally, the extension supports and defaults to the config for the existing extension. This is so it can be a drop in replacement for the existing extension with an easy migration path to the new functionality and features introduced the v2 extension. See the [parent PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30673 ) to understand how the pieces fit together.

**Link to tracking Issue:** #26661

**Testing:** Units / manual

**Documentation:** Readme, etc.

### CODE REVIEW

**Review by codeboten — COMMENTED** [review]
Thanks @mwear, just a few comments

File: extension/healthcheckv2extension/README.md [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+Health Check Extension V1 enables an HTTP url that can be probed to check the
+status of the OpenTelemetry Collector. This extension can be used as a
+liveness and/or readiness probe on Kubernetes.
+
+The following settings are required:
+
+- `endpoint` (default = 0.0.0.0:13133): Address to publish the health check status. For full list of `ServerConfig` refer [here](https://github.com/open-telemetry/opentelemetry-collector/tree/main/config/confighttp). The `component.UseLocalHostAsDefaultHost` feature gate changes this to localhost:13133. This will become the default in a future release.
+- `path` (default = "/"): Specifies the path to be configured for the health check server.
+- `response_body` (default = ""): Specifies a static body that overrides the default response returned by the health check service.
+- `check_collector_pipeline:` (deprecated and ignored): Settings of collector pipeline health check
```
[the remark this answers is not in the document]
Comment (mwear) [reply extension/healthcheckv2extension/README.md:33]: The intention is for this extension to be a seamless replacement for v1. To that end, it supports, and defaults to the v1 configuration. Users need to explicitly opt-in to v2 behavior at this stage. I think the documentation should remain until v1 compatibility is phased out.

[the remark this answers is not in the document]
Comment (codeboten) [reply extension/healthcheckv2extension/README.md:33]: @mwear fair enough, i was hoping we'd correct the mistakes of the past here, but it's not necessary

**Review by codeboten — APPROVED** [review]
Thanks for all the work @mwear

File: extension/healthcheckv2extension/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mwear) [reply extension/healthcheckv2extension/README.md:33]: I think this is the first thing we should remove as we begin to migrate users away from v1 to v2.

**Review by evan-bradley — APPROVED** [review]
Thanks @mwear!

**Pull request merged**